### PR TITLE
Expose resolver via API

### PR DESCRIPTION
**Pull request by sotho**
* Add a new field to the API to expose the "resolver" of a comment/
  conversation. This is not available in the GitHub API v3.
* Extend struct to contain "Resolver". Might be empty (nil).
* Rename "Reviewer" to "Poster" in PullReviewComment to make it
  clear this is the person posting a comment. The API is unchanged
  ('user').
* Only the first comment of a conversation might have a resolver,
  the others seem to be always nil.

### CODE REVIEW

**Comment by richmahn** [comment]
@sotho build is failing. Not sure why just on pgsql though.

**Comment by 6543** [comment]
@sotho please ajust / resolve conflicts - as per #15144

File: modules/convert/pull_review.go [hunk 1]
```diff
@@ -88,14 +88,19 @@ func ToPullReviewCommentList(review *models.Review, doer *models.User) ([]*api.P
...
+				}
+				authResolver := false
+				if doer != nil {
+					authResolver = doer.IsAdmin || (comment.ResolveDoer != nil && doer.ID == comment.ResolveDoer.ID)
 				}
 				apiComment := &api.PullReviewComment{
 					ID:           comment.ID,
 					Body:         comment.Content,
-					Reviewer:     ToUser(comment.Poster, doer != nil, auth),
+					Poster:       ToUser(comment.Poster, doer != nil, authPoster),
```
Comment (zeripath) [inline modules/convert/pull_review.go]: This is a breaking change and doesn't align us with github's api which has this as user (AFAICS)

Comment (6543) [reply modules/convert/pull_review.go]: It's not see PullReviewComment struct

Comment (zeripath) [reply modules/convert/pull_review.go]: I see! Sorry.

**Review by zeripath — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Pull request merged**